packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts

```diff
@@ -407,6 +423,41 @@ export class UserPoolClient extends Resource implements IUserPoolClient {
     return this._userPoolClientName;
   }
 
+  public get userPoolClientSecret(): string | undefined {
+    if (!this._generateSecret) {
+      throw new Error(
+        'userPoolClientSecret is available only if the "generateSecret" prop is set to true',
+      );
+    }
+
+    // Create the Custom Resource that assists in resolving the User Pool Client secret
+    // just once, no matter how many times this method is called
+    if (!this._describeCognitoUserPoolClientSecret) {
+      this._describeCognitoUserPoolClientSecret = new AwsCustomResource(
+        this,
+        'DescribeCognitoUserPoolClient',
+        {
+          resourceType: 'Custom::DescribeCognitoUserPoolClient',
+          onCreate: {
+            region: Stack.of(this).region,
+            service: 'CognitoIdentityServiceProvider',
+            action: 'describeUserPoolClient',
+            parameters: {
+              UserPoolId: this.userPool.userPoolId,
+              ClientId: this.userPoolClientId,
+            },
+            physicalResourceId: PhysicalResourceId.of(this.userPoolClientId),
+          },
+          policy: AwsCustomResourcePolicy.fromSdkCalls({
+            resources: AwsCustomResourcePolicy.ANY_RESOURCE,
```

Comment:
```suggestion
            resources: [this.userPool.userPoolArn],
```

Reply:
IAM policy restricted in scope as per the suggestion.